packages/core/src/config/types.ts

```diff
@@ -195,7 +196,10 @@ interface AbstractActionInput extends SphinxTransaction {
  * Config object with added compilation details. Must add compilation details to the config before
  * the config can be published or off-chain tooling won't be able to re-generate the deployment.
  */
-export interface CompilerConfig extends ParsedConfig {
+export interface DeploymentConfig {
+  networkConfigs: Array<NetworkConfig>
+  merkleTree: SphinxMerkleTree
+  configArtifacts: ConfigArtifacts
```

Comment:
Just want to mention there's a lot of redundancy here b/c the `configArtifacts` maps each fully qualified name to a separate copy of the build info. Could make this pretty large when there are a lot of unique contracts in the deployment. (I know that we trimmed the size of the build info significantly). Still, could the size of this be problematic when uploading to aws?

Reply:
> We could reduce the size of the file we're uploading to our backend further by avoiding duplication of the build info file (since we include one for every contract artifact in the ConfigArtifacts type). I chose not to do that as part of this PR, though, since this is already quite large. We don't technically need to do it since uploading files directly to s3 allows us to get away with having very large files. However, since it would be a breaking change, and it's not that hard, I'll probably go ahead and do that before releasing this. I'm just going to do it separately from this PR.